source/common/http/http1/codec_impl.h

```diff
@@ -215,13 +215,32 @@ class ConnectionImpl : public virtual Connection, protected Logger::Loggable<Log
   void onUnderlyingConnectionAboveWriteBufferHighWatermark() override { onAboveHighWatermark(); }
   void onUnderlyingConnectionBelowWriteBufferLowWatermark() override { onBelowLowWatermark(); }
 
+  // Codec errors found in callbacks are overridden within the http_parser library. This holds those
+  // errors to propagate them through to dispatch() where we can handle the error.
+  Envoy::Http::Status codec_status_;
+
 protected:
   ConnectionImpl(Network::Connection& connection, Stats::Scope& stats, http_parser_type type,
                  uint32_t max_headers_kb, const uint32_t max_headers_count,
                  HeaderKeyFormatterPtr&& header_key_formatter, bool enable_trailers);
 
   bool resetStreamCalled() { return reset_stream_called_; }
 
+  // The following define special return values for http_parser callbacks.
```

Comment:
put in the comment here where these constants come from, and maybe that they need to not overlap with standard HTTP Status codes?

Reply:
done

Reply:
Thanks -- one clarification: you said "These codes do not overlap with standard HTTP Status codes". Is it also correct to strengthen that to "These codes *must* not overlap with standard HTTP Status codes"? Or are they unrelated?

Reply:
Unrelated. They're return values for the user defined callbacks for the parser, so they don't do anything but signal execution flow for the parser.

Reply:
Maybe "The use of these codes are distinct from standard HTTP Status codes"